ksqldb-rest-app/src/main/java/io/confluent/ksql/logging/query/QueryLogger.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ * http://www.confluent.io/confluent-community-license
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logging.query;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.ksql.util.KsqlConfig;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.apache.log4j.Appender;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.log4j.rewrite.RewriteAppender;
+
+public final class QueryLogger {
+  private static final Logger logger = LogManager.getLogger(QueryLogger.class);
+  private static final RewriteAppender rewriteAppender = new RewriteAppender();
+
+  private QueryLogger() {}
+
+  public static void initialize() {
+    logger.setAdditivity(false);
+
+    final Enumeration enumeration = logger.getAllAppenders();
+    final ArrayList<Appender> list = Collections.list(enumeration);
+    logger.addAppender(rewriteAppender);
+    for (Appender a : list) {
+      if (a != rewriteAppender) {
+        logger.removeAppender(a);
+        rewriteAppender.addAppender(a);
+      }
+    }
+  }
+
+  public static void addAppender(final Appender appender) {
+    logger.addAppender(appender);
+  }
+
+  public static Logger getLogger() {
+    return logger;
+  }
+
+  public static void configure(final KsqlConfig config) {
+    rewriteAppender.setRewritePolicy(new QueryAnonymizingRewritePolicy(config));
+  }
+
+  private static ImmutableMap<String, Object> buildPayload(
```

Comment:
Seems like it would be better to use a stronger type (via a POJO) than a map here. I know we have tests to cover usage of this map (so the guarantees the compiler gives you aren't wins from a correctness pov per se), but it also makes things less error-prone for future devs. Plus it's more idiomatic.